lib/puppet/provider/package/yum.rb

```diff
@@ -159,6 +172,11 @@ def install
         self.debug "Downgrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
         operation = :downgrade
       end
+
+      if is && rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure])) > 0
+        self.debug "Updrading package #{@resource[:name]} from version #{is[:ensure]} to #{should}"
```

Comment:
And a note on this debug message - does this log information that isn't already being logged by the resource harness/transaction?